pkg/utils/o11y/controller.go

```diff
@@ -77,23 +86,100 @@ func (h *SuiteController) GetRegexPodNameWithSize(podNameRegex string, results [
 					return nil, fmt.Errorf("error parsing value for %s: %s", podName, err)
 				}
 				valueInMegabytes := h.convertBytesToMB(valueInBytes)
-				podNameWithSize[podName] = valueInMegabytes
+				podNameWithResult[podName] = int(valueInMegabytes)
 			}
 		}
 	}
 
-	if len(podNameWithSize) == 0 {
+	if len(podNameWithResult) == 0 {
 		return nil, fmt.Errorf("no pods matching the regex pattern were found")
 	}
 
-	return podNameWithSize, nil
+	return podNameWithResult, nil
+}
+
+func labelsToSelector(labelMap map[string]string) labels.Selector {
+	selector := labels.NewSelector()
+	for key, value := range labelMap {
+		req, _ := labels.NewRequirement(key, selection.Equals, []string{value})
+		selector = selector.Add(*req)
+	}
+	return selector
+}
+
+func (s *SuiteController) WaitForScriptCompletion(deployment *appsv1.Deployment, successMessage string, timeout time.Duration) error {
+	namespace := deployment.Namespace
+	deploymentName := deployment.Name
+
+	// Get the pod associated with the deployment
+	podList := &corev1.PodList{}
+	labels := deployment.Spec.Selector.MatchLabels
+	labelSelector := labelsToSelector(labels)
+	err := s.KubeRest().List(context.Background(), podList, client.InNamespace(namespace), client.MatchingLabelsSelector{Selector: labelSelector})
+	if err != nil {
+		return err
+	}
+
+	if len(podList.Items) == 0 {
+		return fmt.Errorf("no pods found for deployment %s", deploymentName)
+	}
+
+	pod := podList.Items[0]
+
+	// Wait for the success message in the pod's log output
+	podLogOpts := &corev1.PodLogOptions{}
+	req := s.KubeInterface().CoreV1().Pods(namespace).GetLogs(pod.Name, podLogOpts)
+
+	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		readCloser, err := req.Stream(context.Background())
+		if err != nil {
+			return false, err
+		}
+		defer readCloser.Close()
+
+		scanner := bufio.NewScanner(readCloser)
+		for scanner.Scan() {
+			if strings.Contains(scanner.Text(), successMessage) {
+				return true, nil
+			}
+		}
+		return false, nil
```

Comment:
Shouldn't return an error informing that the script did not finish?

Reply:
for timeout. Yes

Reply:
Done